squarelet/users/views.py

```diff
@@ -148,22 +158,65 @@ class UserListView(LoginRequiredMixin, ListView):
 
 
 class LoginView(AllAuthLoginView):
-    """Subclass of All Auth Login View to add redirect ability for failed auth tokens
-
-    If the url_auth_token parameter is still present, it means the auth token failed
-    to authenticate the user.  Redirect them to the nested next parameter instead of
-    asking them to login
-    """
+    def dispatch(self, request, *args, **kwargs):
+        """Override form_valid to provide onboarding steps, if needed"""
+        # handle the request as usual, but hold onto the response
+        response = super().dispatch(request, *args, **kwargs)
+        # when logging in, a POST is sent
+        is_post = request.method == "POST"
```

Comment:
Since you only care about POST, would it be cleaner to override the `post` method instead of the `dispatch` method?

Reply:
I tried that initially, but found that the redirect URL was incorrect during testing. After digging into it, I believe this is because of how AllAuth's `RedirectAuthenticatedUserMixin` overrides dispatch. In order to get the right redirect URL, we need to handle it inside `dispatch`.